### PR TITLE
[WIP] [egs] port tdnn_7m23t script to librispeech

### PR DESCRIPTION
**Pull request by davidavdav**
This is the port of the tdnn_7m23t.sh script from swbd to librispeech.  It took a while, since I wanted to verify all steps actually run.  Still I added one/two things afterwards (e.g., the num_leaves parameter).  I didn't make it to PR2114 in time, so therefore this is just a PR to master.  

Results measured are:
```
for d in exp/chain/tdnn7m23t_sp/decode_*; do ../bin/best $d; done
exp/chain/tdnn7m23t_sp/decode_dev_clean_fglarge/wer_12_0.5:%WER 3.40 [ 1851 / 54402, 229 ins, 155 del, 1467 sub ]
exp/chain/tdnn7m23t_sp/decode_dev_clean_tglarge/wer_12_1.0:%WER 3.51 [ 1910 / 54402, 202 ins, 199 del, 1509 sub ]
exp/chain/tdnn7m23t_sp/decode_dev_clean_tgmed/wer_12_0.0:%WER 4.28 [ 2327 / 54402, 265 ins, 206 del, 1856 sub ]
exp/chain/tdnn7m23t_sp/decode_dev_clean_tgsmall/wer_12_0.0:%WER 4.84 [ 2635 / 54402, 277 ins, 263 del, 2095 sub ]
exp/chain/tdnn7m23t_sp/decode_dev_other_fglarge/wer_14_0.5:%WER 8.77 [ 4468 / 50948, 484 ins, 500 del, 3484 sub ]
exp/chain/tdnn7m23t_sp/decode_dev_other_tglarge/wer_14_0.0:%WER 9.24 [ 4709 / 50948, 598 ins, 435 del, 3676 sub ]
exp/chain/tdnn7m23t_sp/decode_dev_other_tgmed/wer_14_0.0:%WER 11.31 [ 5760 / 50948, 604 ins, 678 del, 4478 sub ]
exp/chain/tdnn7m23t_sp/decode_dev_other_tgsmall/wer_14_0.0:%WER 12.45 [ 6342 / 50948, 596 ins, 829 del, 4917 sub ]
exp/chain/tdnn7m23t_sp/decode_test_clean_fglarge/wer_11_0.5:%WER 3.87 [ 2036 / 52576, 317 ins, 157 del, 1562 sub ]
exp/chain/tdnn7m23t_sp/decode_test_clean_tglarge/wer_10_0.5:%WER 4.04 [ 2126 / 52576, 336 ins, 164 del, 1626 sub ]
exp/chain/tdnn7m23t_sp/decode_test_clean_tgmed/wer_12_0.0:%WER 4.90 [ 2575 / 52576, 345 ins, 229 del, 2001 sub ]
exp/chain/tdnn7m23t_sp/decode_test_clean_tgsmall/wer_12_0.0:%WER 5.35 [ 2811 / 52576, 340 ins, 269 del, 2202 sub ]
exp/chain/tdnn7m23t_sp/decode_test_other_fglarge/wer_13_0.5:%WER 8.97 [ 4694 / 52343, 564 ins, 481 del, 3649 sub ]
exp/chain/tdnn7m23t_sp/decode_test_other_tglarge/wer_13_0.5:%WER 9.42 [ 4931 / 52343, 581 ins, 542 del, 3808 sub ]
exp/chain/tdnn7m23t_sp/decode_test_other_tgmed/wer_13_0.0:%WER 11.56 [ 6053 / 52343, 690 ins, 659 del, 4704 sub ]
exp/chain/tdnn7m23t_sp/decode_test_other_tgsmall/wer_13_0.0:%WER 12.64 [ 6615 / 52343, 678 ins, 791 del, 5146 sub ]

for d in exp/chain/tdnn7m23t_sp_online/decode_*; do ../bin/best $d; done
exp/chain/tdnn7m23t_sp_online/decode_dev_clean_tgsmall/wer_12_0.0:%WER 4.83 [ 2630 / 54402, 272 ins, 262 del, 2096 sub ]
exp/chain/tdnn7m23t_sp_online/decode_dev_other_tgsmall/wer_14_0.0:%WER 12.52 [ 6381 / 50948, 598 ins, 828 del, 4955 sub ]
exp/chain/tdnn7m23t_sp_online/decode_test_clean_tgsmall/wer_12_0.0:%WER 5.33 [ 2802 / 52576, 343 ins, 265 del, 2194 sub ]
exp/chain/tdnn7m23t_sp_online/decode_test_other_tgsmall/wer_13_0.0:%WER 12.67 [ 6630 / 52343, 657 ins, 799 del, 5174 sub ]
```

### CODE REVIEW

File: egs/librispeech/s5/local/chain/tuning/run_tdnn_7m23t.sh [hunk 1]
```diff
@@ -0,0 +1,246 @@
+#!/bin/bash
+
+## Adapted from swbd for librispeech by David van Leeuwen
```
Comment (danpovey) [inline egs/librispeech/s5/local/chain/tuning/run_tdnn_7m23t.sh]: Would you mind redoing this based on the checked-in librispeech example in master?  (I think it's 7n or something like that).  I renamed some of the layers in a way that I consider clearer.  And rename to run_tdnn_1c.sh and change the suffix to 1c; Also we need a comparison with the old results, produced by compare_wer.sh.  It's betterif you actually run the old setup (in 1b) and compare with that; if not, you could try to fake it based on its compare_wer.sh output in its comment at the top.

Comment (davidavdav) [reply egs/librispeech/s5/local/chain/tuning/run_tdnn_7m23t.sh]: Sure, no problem I can do that.  All it takes is time, since I have a small cluster.

**Comment by davidavdav** [comment]
I re-modeled the body of the script to `run_tdnn_1b.sh` on master, and included a compare_wer.sh for (my version) of 1b and (now renamed to) 1c.  

Don't merge yet---I still need to 
 - re-run 1b to check dir structure and performance
 - copy nnet config from swbd 7n into this 1c
 - re-run 1c for performance comparison.

**Comment by davidavdav** [comment]
You're OK to merge now.  Re-ran the `tdnn_1c` nnet training, and inserted results into the top of the script, comparing to `tdnn_1b`.  

I can rebase and change subject to `[egs] port tdnn_7m23t script to librispeech` if you wish.

**Comment by danpovey** [comment]
Thanks!  Merging.

**Pull request merged**